apps/client-server/src/app/websites/implementations/bluesky/bluesky.website.ts

```diff
@@ -0,0 +1,698 @@
+import {
+  $Typed,
+  AppBskyActorGetProfile,
+  AppBskyEmbedImages,
+  AppBskyEmbedVideo,
+  AppBskyFeedThreadgate,
+  AppBskyVideoGetJobStatus,
+  AppBskyVideoGetUploadLimits,
+  AppBskyVideoUploadVideo,
+  AtpAgent,
+  AtUri,
+  BlobRef,
+  ComAtprotoLabelDefs,
+  RichText,
+} from '@atproto/api';
+import { ReplyRef } from '@atproto/api/dist/client/types/app/bsky/feed/post';
+import { JobStatus } from '@atproto/api/dist/client/types/app/bsky/video/defs';
+import {
+  BlueskyAccountData,
+  BlueskyOAuthRoutes,
+  FileType,
+  ILoginState,
+  ImageResizeProps,
+  ISubmissionFile,
+  OAuthRouteHandlers,
+  PostData,
+  PostResponse,
+  SimpleValidationResult,
+  SubmissionRating,
+  SubmissionValidator,
+} from '@postybirb/types';
+import { getFileTypeFromMimeType } from '@postybirb/utils/file-type';
+import { CancellableToken } from '../../../post/models/cancellable-token';
+import { PostingFile } from '../../../post/models/posting-file';
+import { CustomLoginFlow } from '../../decorators/login-flow.decorator';
+import { SupportsFiles } from '../../decorators/supports-files.decorator';
+import { SupportsUsernameShortcut } from '../../decorators/supports-username-shortcut.decorator';
+import { WebsiteMetadata } from '../../decorators/website-metadata.decorator';
+import { DataPropertyAccessibility } from '../../models/data-property-accessibility';
+import { FileWebsite } from '../../models/website-modifiers/file-website';
+import { MessageWebsite } from '../../models/website-modifiers/message-website';
+import { OAuthWebsite } from '../../models/website-modifiers/oauth-website';
+import { Website } from '../../website';
+import { BlueskyFileSubmission } from './models/bluesky-file-submission';
+import { BlueskyMessageSubmission } from './models/bluesky-message-submission';
+
+@WebsiteMetadata({
+  name: 'bluesky',
+  displayName: 'BlueSky',
+})
+@CustomLoginFlow()
+@SupportsUsernameShortcut({
+  id: 'bsky',
+  url: 'https://bsly.app/profile/$1',
+})
+@SupportsFiles({
+  acceptedMimeTypes: [
+    'image/png',
+    'image/jpeg',
+    'image/gif',
+    'video/mp4',
+    'video/mov',
+    'video/webm',
+  ],
+  acceptedFileSizes: { '*': 1_000_000 },
+  fileBatchSize: 4,
+})
+export default class Bluesky
```

Comment:
thought: I think we might just want to add the @DisableAds() decorator to all websites that have low description length limits.